code/lab/lab.cpp

```diff
@@ -2093,32 +2097,41 @@ void lab_weapon_fire_window_close(GUIObject* /*caller*/) {
 	Lab_weapon_fire_window = nullptr;
 }
 
+void lab_animations_window_close(GUIObject* /*caller*/) {
+	Lab_trigger_animations_window = nullptr;
+}
+
 void labviewer_actions_destroy_ship(Button* /*caller*/) {
-	auto obj = &Objects[Lab_selected_object];
+	if (Lab_selected_object != -1) {
+		auto obj = &Objects[Lab_selected_object];
 
-	if (obj->type == OBJ_SHIP) {
-		obj->flags.remove(Object::Object_Flags::Player_ship);
-		ship_self_destruct(obj);
-		Lab_viewer_flags |= LAB_FLAG_SHIP_EXPLODING;
+		if (obj->type == OBJ_SHIP) {
+			obj->flags.remove(Object::Object_Flags::Player_ship);
+			ship_self_destruct(obj);
+			Lab_viewer_flags |= LAB_FLAG_SHIP_EXPLODING;
+		}
 	}
 }
 
+// Returns true if it is safe to access the Ships array
+bool actions_window_is_safe() {
```

Comment:
Call the function `actions_window_safe_for_ships` or something for future proofing.  There might be non-ship actions in the future.